cli/src/pcluster/templates/login_nodes_stack.py

```diff
@@ -0,0 +1,229 @@
+        self._shared_storage_mount_dirs = shared_storage_mount_dirs
+        self._shared_storage_attributes = shared_storage_attributes
+
+        self._add_resources()
+
+    @property
+    def stack_name(self):
+        """Name of the CFN stack."""
+        return Stack.of(self.nested_stack_parent).stack_name
+
+    def _add_resources(self):
+        # get unique login nodes pools subnet id
+        subnet_ids = set(pool.networking.subnet_id for pool in self._login_nodes.pools)
+        self._pools_subnet_ids = list(subnet_ids)
+
+        # get unique login nodes pools availability_zone
+        availability_zones = set(pool.networking.availability_zone for pool in self._login_nodes.pools)
+        self._pools_availability_zones = list(availability_zones)
+
+        self._vpc = ec2.Vpc.from_vpc_attributes(
+            self,
+            "VPC",
+            vpc_id=self._config.vpc_id,
+            availability_zones=self._pools_availability_zones,
+        )
+        self._login_nodes_target_group = self._add_login_nodes_target_group()
+        self._login_nodes_load_balancer = self._add_login_nodes_load_balancer(self._login_nodes_target_group)
+
+        self.pools = {}
+        for pool in self._login_nodes.pools:
+            pool_construct = Pool(
+                self,
+                f"Pool{pool.name}",
+                pool,
+                self._config,
+                self._shared_storage_infos,
+                self._shared_storage_mount_dirs,
+                self._shared_storage_attributes,
+                self._login_security_group,
+                self._login_nodes_target_group,
+                self.stack_name,
+            )
+            self.pools[pool.name] = pool_construct
+
+    def _add_login_nodes_target_group(self):
+        return elbv2.NetworkTargetGroup(
+            self,
+            f"{self.stack_name}-TargetGroup",
+            health_check=elbv2.HealthCheck(
+                port="22",
+                protocol=elbv2.Protocol.TCP,
+            ),
+            port=22,
+            protocol=elbv2.Protocol.TCP,
+            target_group_name=f"{self.stack_name}-TargetGroup",
+            target_type=elbv2.TargetType.INSTANCE,
+            vpc=self._vpc,
+        )
+
+    def _add_login_nodes_load_balancer(
+        self,
+        target_group,
+    ):
+        login_nodes_load_balancer = elbv2.NetworkLoadBalancer(
+            self,
+            f"{self.stack_name}-LoadBalancer",
+            vpc=self._vpc,
+            internet_facing=True,
+            vpc_subnets=ec2.SubnetSelection(
+                subnets=[
+                    ec2.Subnet.from_subnet_id(self, f"LoginNodesSubnet{i}", subnet_id)
+                    for i, subnet_id in enumerate(self._pools_subnet_ids)
+                ]
+            ),
+        )
+
+        listener = login_nodes_load_balancer.add_listener("LoginNodeListener", port=22)
+        listener.add_target_groups("LoginNodeListenerTargets", target_group)
```

Comment:
Same question, this listener is associated with a single pool. The `id` should avoid conflicts.

Reply:
Done! I format the id with pool.name.